extensions/resteasy-classic/resteasy-client/deployment/src/test/java/io/quarkus/restclient/exception/SubResourceTest.java

```diff
@@ -0,0 +1,165 @@
+            fail("fromSub() should have thrown a TestException");
+        } catch (TestException expected) {
+            assertEquals("SubResource failed on purpose", expected.getMessage());
+        }
+    }
+
+    /**
+     * This test invokes a call to the sub-resource. The sub-resource then invokes an additional call which should
+     * return the header value for {@code test-header}.
+     *
+     */
+    @Test
+    public void subResourceWithHeader() {
+        try (final Response response = clientRootResource.subResource().withHeader()) {
+            assertEquals(OK, response.getStatusInfo());
+            assertEquals("SubResourceHeader", response.readEntity(String.class));
+        }
+    }
+
+    /**
+     * This test invokes a call to the sub-resource. The sub-resource then invokes an additional call which should
+     * return the header value for {@code test-global-header}.
+     *
+     * @throws Exception if a test error occurs
+     */
+    @Test
+    public void subResourceWithGlobalHeader() throws Exception {
+        try (final Response response = clientRootResource.subResource().withGlobalHeader()) {
+            assertEquals(OK, response.getStatusInfo());
+            assertEquals("GlobalSubResourceHeader", response.readEntity(String.class));
+        }
+    }
+
+    @RegisterRestClient
+    @RegisterProvider(TestExceptionMapper.class)
+    @Path("/root")
+    public interface ClientRootResource {
+        @Path("/sub")
+        ClientSubResource subResource();
+
+        @GET
+        Response fromRoot() throws TestException;
+    }
+
+    @ClientHeaderParam(name = "test-global-header", value = "GlobalSubResourceHeader")
+    @Produces(TEXT_PLAIN)
+    public interface ClientSubResource {
+        @GET
+        Response fromSub() throws TestException;
+
+        @GET
+        @ClientHeaderParam(name = "test-header", value = "SubResourceHeader")
+        @Path("/header")
+        Response withHeader();
+
+        @GET
+        @Path("/global/header")
+        Response withGlobalHeader();
+    }
+
+    @Path("/root")
+    public static class ServerResource {
+        @GET
+        public Response fromRoot() {
+            return Response.serverError().entity("RootResource failed on purpose").build();
+        }
+
+        @GET
+        @Path("/sub")
+        public Response fromSub() {
+            return Response.serverError().entity("SubResource failed on purpose").build();
+        }
+
+        @GET
+        @Path("/sub/header")
+        public Response subHeader(@HeaderParam("test-header") final String value) {
+            return Response.ok(value).build();
+        }
+
+        @GET
+        @Path("/sub/global/header")
+        public Response subGlobalHeader(@HeaderParam("test-global-header") final String value) {
+            return Response.ok(value).build();
+        }
+    }
+
+    public static class TestExceptionMapper implements ResponseExceptionMapper<TestException> {
```

Comment:
Does this need a CDI qualifying annotation?

Reply:
will annotate it as 'Singleton' since it is also registered as @Provider it should be fine either way

Reply:
There is no annotation needed here but it doesn't really matter that it is present.